src/Features/LanguageServer/Protocol/Handler/CodeLens/CodeLensCache.cs

```diff
@@ -10,7 +10,7 @@
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CodeLens;
 
-internal sealed class CodeLensCache : ResolveCache<CodeLensCacheEntry>
+internal sealed class CodeLensCache : ResolveCache<CodeLensCacheEntry, string>
```

Comment:
could you just store the SyntaxVersion itself here, not the ToString of it?

Reply:
Basically to make passing this to and from the client easy. The [code lens request ](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_codeLens) data object is [LSPAny](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#lspAny). I could make the VersionStamp a LSPObject but I wasn't sure how that worked exactly with LSP and passing the whole object didn't seem high enough priority. Thoughts?